compiler/hash-typecheck/src/traits.rs

```diff
@@ -1,19 +1,47 @@
+//! Traits for typechecking on TIR nodes.
+//!
+//! This defines a structured way to implement the typechecker, where
+//! each node in the TIR `X` has a corresponding trait implementation
+//! `Operations<X>` which defines how to typecheck, unify, and normalise that
+//! node.
 use std::ops::ControlFlow;
 
 use crate::{env::HasTcEnv, errors::TcResult, options::normalisation::NormaliseResult};
 
+/// Main trait for typechecking on TIR atoms.
+///
+/// `X` is the TIR atom type.
 pub trait Operations<X>: HasTcEnv {
-    type TyNode;
```

Comment:
What if this trait was called `OperationsOn<Atom>` ?

Reply:
Why need `On`?

Reply:
I don't want to use `Atom` because that is a defined type

Reply:
I see, alright